src/itype.h

```diff
@@ -616,6 +616,8 @@ struct islot_ammo : common_ranged_data {
      * @warning It is not read from the json directly.
      * */
     bool special_cookoff = false;
+
+    float prop_damage = -1.0;
```

Comment:
Please use `cata::optional` instead of some arbitrary and undocumented sentinel value:

```C++
cata::optional<float> prop_damage;

// Check that is contains an actual value like this:
if( type->ammo->prop_damage ) {
    // Dereference the object to get the actual value like this:
    real_damage = gun_damage * *type->ammo->prop_damage;
}
```